node/src/components/consensus/protocols/highway.rs

```diff
@@ -125,7 +125,7 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
 
         let init_round_exp = prev_cp
             .and_then(|cp| cp.as_any().downcast_ref::<HighwayProtocol<I, C>>())
-            .and_then(|highway_proto| highway_proto.median_round_exp())
+            .and_then(|highway_proto| highway_proto.our_round_exp())
```

Comment:
Can it happen that we never activated in the previous era? E.g. because we just joined? Or because in the future we make some other change that deactivates the old era _before_ initializing the new one — it's just one more of those implicit assumptions.

Would it be better to not use `active_validator` for this at all, and instead use our validator ID to retrieve the latest unit belonging to that ID?

Reply:
If we never activated in the previous era, then we also haven't created any units, so that part is not a problem in my opinion.

Deactivation before initialization of the new era could be a problem, however. But... isn't `active_validator` the only part that actually knows our validator ID?